frontend/src/metabase/visualizations/components/settings/ChartSettingInputSuggestion.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { useState, useRef } from "react";
+import TippyPopoverWithTrigger from "metabase/components/PopoverWithTrigger/TippyPopoverWithTrigger";
+import SelectList from "metabase/components/SelectList";
+import { useListKeyboardNavigation } from "metabase/hooks/use-list-keyboard-navigation";
+import {
+  SuggestionInput,
+  SuggestionContainer,
+} from "./ChartSettingInputSuggestion.styled";
+
+interface ChartSettingInputSuggestionProps {
+  value: string;
+  onChange: (value: string) => void;
+  id?: string;
+  options?: string[];
+}
+
+const matchPattern = /.*{{([^{}]*)$/;
+
+const ChartSettingInputSuggestion = ({
+  value: initialValue,
+  onChange,
+  options,
+  ...props
+}: ChartSettingInputSuggestionProps) => {
+  const [suggestions, setSuggestions] = useState<string[]>([]);
+  const [value, setValue] = useState(initialValue);
+  const optionsListRef = useRef<HTMLUListElement>(null);
+  const inputRef = useRef<HTMLDivElement>(null);
+
+  const { cursorIndex } = useListKeyboardNavigation({
```

Comment:
Keyboard navigation would be useful for all Select lists. Wdyt about moving the keyboard navigation there? SelectList could accept a render function as `children` prop so that the API would look like this:
```
<SelectList items={items}>
  {(items, selectedId} => <>
     <SelectList.Item>....
  </>}
</SelectList>
```